pkg/cmd/secret/set/set.go

```diff
@@ -209,15 +214,26 @@ func setRun(opts *SetOptions) error {
 		}
 	}()
 
+	secretEntity := shared.GetSecretEntity(orgName, envName, opts.UserSecrets)
+	secretApp := shared.GetSecretApp(opts.Application, secretEntity)
+	if secretApp == shared.Unknown {
+		return fmt.Errorf("invalid application: %s", opts.Application)
+	}
+
+	if !shared.IsSupportedSecretEntity(secretApp, secretEntity) {
+		return fmt.Errorf("%s secrets are not supported for %s", secretEntity, secretApp)
+	}
```

Comment:
Since this is input validation I think it would be nice to move these checks above the `mapRepoNamesToIDs` code so that we don't start making API requests before we know they will be necessary.

Reply:
https://github.com/cli/cli/pull/5134/commits/55c029388bb2459a582be10c6cd180db59f50d3f